lib/src/body.rs

```diff
@@ -199,6 +206,38 @@ impl HttpBody for Body {
                         }
                     }
                 }
+                Chunk::CompressedHttpBody(mut decoder_state, mut body) => {
+                    let body_mut = &mut body;
+                    pin_mut!(body_mut);
+
+                    match body_mut.poll_data(cx) {
+                        Poll::Pending => {
+                            // put the body back, so we can poll it again next time
+                            self.chunks.push_front(body.into());
+                            return Poll::Pending;
+                        }
+                        Poll::Ready(None) => {
+                            match decoder_state.finish() {
+                                Err(e) => return Poll::Ready(Some(Err(e.into()))),
+                                Ok(buffer) => return Poll::Ready(Some(Ok(bytes::Bytes::from(buffer)))),
+                            }
+                        }
+                        Poll::Ready(Some(item)) => {
+                            // put the body back, so we can poll it again next time
+                            self.chunks.push_front(body.into());
```

Comment:
I think we need to push the entire `CompressedHttpBody` chunk to the front of the queue, to continue reading into it.

Reply:
Ah, yes. Good catch. Should be addressed in 88146fbeba037d351d16729c0b7d03c2ee982a67.